src/Streamly/Internal/Data/Stream/StreamD/Eliminate.hs

```diff
@@ -206,20 +217,13 @@ parse_ (PRD.Parser pstep initial extract) (Stream step state) = do
                 let (src0, buf1) = splitAt n (x:getList buf)
                     src  = Prelude.reverse src0 ++ xs
                 gobuf SPEC s (List buf1) (List src) pst1
-            PR.Done _ b -> return (b, Stream step_rest (Left (s, buf)))
+            PR.Done n b -> do
```

Comment:
Have a `Done 0` case here as well?

Reply:
Yes, that would avoid an unnecessary `append` which is detrimental to perf.

Reply:
the `Done 0` case won't be very useful here as the append would still be needed. And the rest of the code like `take` would anyway be simplified. So I am keeping it as is.